sinex/src/datetime.rs

```diff
@@ -17,7 +17,7 @@ pub fn parse_datetime(content: &str) -> Result<chrono::NaiveDateTime, ParseDateT
     let h = secs / 3600.0;
     let m = (secs - h * 3600.0) / 60.0;
     let s = secs - h * 3600.0 - m * 60.0;
-    Ok(dt.and_hms(h as u32, m as u32, s as u32))
+    Ok(dt.and_hms_opt(h as u32, m as u32, s as u32).unwrap())
```

Comment:
```rust
Ok(dt.and_hms_opt(h as u32, m as u32, s as u32)?)
```

in relation to `ParseDateTimeError`

Reply:
Done now :)

Reply:
But this fails with `use .ok_or(...)? to provide an error compatible with `Result<NaiveDateTime, ParseDateTimeError>`

Reply:
you can either use thiserror::Error to provide automated casts from chrono::Errors to ParseDateTimeError, 
or else need to cast yourself, with .ok_or(Err(Error::SomeKindOfError)).

you can find similar scenarios in the rinex/src/epoch or rinex/src/header modules

I would not recommend spending time on this, this crate will eventually get updated to use Hifitime